xbmc/guilib/GUIControl.cpp

```diff
@@ -124,21 +124,25 @@ void CGUIControl::DynamicResourceAlloc(bool bOnOff)
 // 1. animate and set animation transform
 // 2. if visible, process
 // 3. reset the animation transform
-void CGUIControl::DoProcess(unsigned int currentTime, CDirtyRegionList &dirtyregions)
+void CGUIControl::DoProcess(CGUIRenderInfo &renderInfo)
 {
   CRect dirtyRegion = m_renderRegion;
 
   bool changed = m_bInvalidated && IsVisible();
 
-  changed |= Animate(currentTime);
+  changed |= Animate(renderInfo.GetTime());
 
+  if (m_controlID != WINDOW_FULLSCREEN_VIDEO)
```

Comment:
I dont like to have this special case in CGUIControl, its against OOP.
It should be solved using virtual member (IncVisibleCount()) in CGUIControl and should be overwritten in CGUIWindowFullScreen.